docroot/profiles/custom/cgov_site/modules/custom/cgov_site_section/cgov_site_section.module

```diff
@@ -119,35 +164,178 @@ function cgov_site_section_taxonomy_term_update(EntityInterface $entity) {
-  // issue regarding workflow and the various entity types we would have to hit.
+/**
+ * Determines if a section should trigger a re-save of its children terms.
+ *
+ * @param \Drupal\Core\Entity\EntityInterface $updated_section_entity
+ *   The site section entity with its pending value updates.
+ *
+ * @return bool
+ *   A TRUE or FALSE as to whether to trigger a re-save.
+ */
+function will_trigger_save_children(EntityInterface $updated_section_entity) {
+
+  $fields = [
+    'computed_path',
+    'parent',
+    'field_pretty_url',
+    'langcode',
+  ];
+  return entity_fields_modified($fields, $updated_section_entity);
+}
+
+/**
+ * Determine if an entity should trigger a cache tag invalidation.
+ *
+ * Fields which trigger a cache tag invalidation for a section when modified:
+ *  name
+ *  field_landing_page
+ *  field_pretty_url
+ *  field_section_nav_root
+ *  field_navigation_label
+ *  field_navigation_display_options
+ *  weight.
+ *
+ * For a node or media item:
+ * field_site_section
+ *
+ * @param \Drupal\Core\Entity\EntityInterface $updated_entity
+ *   The entity just updated.
+ *
+ * @return bool
+ *   A TRUE or FALSE as to whether to trigger a cache clear.
+ */
+function _will_trigger_cache_tag_invalidation(EntityInterface $updated_entity) {
+  $values = [];
+
+  // Invalidation Check For Nodes/Media.
+  if ($updated_entity->getEntityTypeId() === 'node' || $updated_entity->getEntityTypeId() === 'media') {
+    // Exit early if this entity doesn't have site sections.
+    if ($updated_entity->hasField('field_site_section')) {
+      return FALSE;
+    }
+    // Set entity Fields which trigger invalidation when modified.
+    $values =
+      [
+        'field_site_section',
+      ];
+  }
+
+  // Invalidation Check for Site Sections.
+  if ($updated_entity->getEntityTypeId() === 'taxonomy_term' &&
+    $updated_entity->bundle() == 'cgov_site_sections') {
+    $weightModified = $updated_entity->original->getWeight() !== $updated_entity->getWeight();
+    // Early exit if weight modified.
+    if ($weightModified) {
+      return TRUE;
+    }
+
+    // Set Site Section Fields which trigger invalidation.
+    $values =
+      [
+        'name',
+        'field_landing_page',
+        'field_pretty_url',
+        'field_section_nav_root',
+        'field_navigation_label',
+        'field_navigation_display_options',
+      ];
+  }
+  // Compare the values and check if any have been modified.
+  return entity_fields_modified($values, $updated_entity);
+}
+
+/**
+ * Checks to see if an entities values have changed compared to it's original.
+ *
+ * @param array $fields
+ *   The fields which trigger a cache invalidation.
+ * @param \Drupal\Core\Entity\EntityInterface $entity
+ *   The updated entity.
+ *
+ * @return bool
+ *   A TRUE or FALSE as to whether the fields were modified.
+ */
+function entity_fields_modified(array $fields, EntityInterface $entity) {
+  // Store the original entity for comparison.
```

Comment:
Compares entity fields in a flexible manner

Reply:
Nice bit of magic here!